js/app.js

```diff
@@ -1,81 +1,424 @@
+                    //console.log(itemObject);
+                    $.each(smplcolor, function(i, el){
+                        if($.inArray(el, itemObject.color) === -1) itemObject.color.push(el);
+                    });
+                    $.each(smplsize, function(i, el){
+                        if($.inArray(el, itemObject.size) === -1) itemObject.size.push(el);
+                    });
+                    $.each(samplcolorurl, function(i, el){
+                        if($.inArray(el, itemObject.colorUrl) === -1) itemObject.colorUrl.push(el);
+                    });
+                    $.each(smplprice, function(i, el){
+                        if($.inArray(el, itemObject.price) === -1) itemObject.price.push(el);
+                    });
+
+                }
+                bundleArray.push(itemObject);
+            }
+            //console.log(bundleArray);
+            vm.Product.jsonData(bundleArray);
+            vm.Product.ratingSort();
+            vm.Product.sizeAll(sizeAll);
+            vm.Product.colorAll(colorAll);
+            vm.Product.brandAll(brandAll);
+            vm.Product.ratingAll(ratingAll);
         });
-        this.get('#About', function () {
-            vm.Main.chosenPageId(this.params.page);     
-            vm.Main.template("page-template")
-            //vm.Page.name("About")
-            $.get('./server/about.json',function(data){
-                vm.Page.name(data.title);
-                vm.Page.content(data.content);
+
+    
+        self.colorHover = function (data, event){
+            //console.log(data[0], data[1])
+            $("#"+data[1].prodId).css('background-image', 'url('+ data[1].colorUrl[data[0]]+')');
+
+            var colorSizes = [];
+            $.each(data[1].productsAll, function(ind, prod){
+                if(prod.color == data[2]){
+                    colorSizes.push(prod.size);
+                }
             })
-        });
-        this.get('#Product', function () {
-            vm.Main.chosenPageId(this.params.page);     
-            vm.Main.template("product-template")
-            //vm.Page.name("About")
-            //fetch product details here
-        });
+            var sizeId = data[1].prodId+'_size'
+            // console.log(colorSizes.toString());
+            $("#"+sizeId).text("Size : "+colorSizes.toString());
 
-        this.get('#Product/:productid', function () {
-            vm.Main.chosenPageId("Product");     
-            vm.Main.template("product-template")
-            //this.params.productid
-            //vm.Page.name("About")
-            //fetch product details here
-        });
+            if(data[1].price[data[0]] != undefined){
+                $(".priceTag"+data[1].prodId).text(' '+data[1].price[data[0]]);
+            }
+
+        }
+
+        self.showAllSizeFilter = function(){
+            // console.log("showAllSizeFilter");
+            vm.Product.sizeFlag(true);
+            $("#showSizeTag").css('display', 'none');
```

Comment:
Avoid changing CSS properties through jquery. Use CSS class to change any property instead in whole file